app/lib/actions/katello/host/hypervisors_update.rb

```diff
@@ -61,6 +62,11 @@ def update_or_create_hypervisor(environment, content_view, hypervisor_json)
 
           host.save!
         end
+
+        def find_host_by_uuid_or_name(hypervisor_json)
+          facet = ::Katello::Host::SubscriptionFacet.find_by(:uuid => hypervisor_json[:uuid])
+          facet.nil? ? ::Host.find_by(:name => hypervisor_json[:name]) : facet.host
```

Comment:
This logic seems to make https://github.com/Katello/katello/pull/6076/files#diff-bf897becee6d218f2e9b589c5f66dcfdR39 redundant in terms of database queries